### PR TITLE
Add CI linting and testing - closes #81

### LINKED ISSUE #81: Create github workflow to run tests
Just to have more security in our PRs, let's make a GitHub Action to run our tests =)

### PR DESCRIPTION
**Pull request by helderburato**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  push:
+    branches:
+      - 'main'
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [12, 14]
```
Comment (willianjusten) [inline .github/workflows/ci.yml:13]: Any reason of using both node versions? I'd use only one, there's no need to run twice, we're not releasing anything that needs to be compatible to different node versions.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,6 @@
-## NBA React
+## NBA Remix
```
Comment (willianjusten) [inline README.md:1]: Oh, thanks for fixing the name here xD

File: package.json [hunk 3]
```diff
@@ -14,7 +14,9 @@
     "test:ci": "jest --runInBand",
     "storybook": "start-storybook -p 6006",
     "build-storybook": "build-storybook",
-    "generate": "plop --plopfile generators/plopfile.js"
+    "generate": "plop --plopfile generators/plopfile.js",
+    "lint": "eslint app --max-warnings=0",
+    "validate": "npm run lint && npm run test:ci"
```
Comment (willianjusten) [inline package.json:19]: I see you creating this but not using anywhere, maybe we should remove it?

**Review by willianjusten — CHANGES_REQUESTED** [review]
Looks good, just adjust some details.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (adeonir) [reply .github/workflows/ci.yml:13]: We can stick with version 14, which is the latest LTS running on Vercel.

File: package.json (hunk 3 above)
Comment (helderburato) [reply package.json:19]: I forgot to remove this one, nice catch!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (helderburato) [reply .github/workflows/ci.yml]: Updated to keep the version `14`.

**Review by willianjusten — APPROVED** [review]
Great =)

**Review by adeonir — APPROVED** [review]

**Pull request merged**